# We begin with classification, so we fit a baseline classifier that predicts the mode

In [ ]:
import pandas as pd
cleaned_data = "../../data/cleaned_cs2_games_pregame.csv"
df = pd.read_csv(cleaned_data, encoding="latin1")
df

,tournament,team1_id,team1,team2_id,team2,bestOf,map_name,team1_win,team1_player1_id,team1_player2_id,...,team1_player5_id,team2_player1_id,team2_player2_id,team2_player3_id,team2_player4_id,team2_player5_id,year,month,day_of_week,hour
0,10,288898,122,288899,199,5.0,3,0,1401.0,169.0,...,7201.0,19.0,550.0,1549.0,1443.0,146.0,2026,3,6,13
1,10,288898,122,288899,199,5.0,1,0,1401.0,169.0,...,7201.0,19.0,550.0,1549.0,1443.0,146.0,2026,3,6,13
2,10,288894,137,288895,128,3.0,3,1,2988.0,266.0,...,16113.0,1401.0,169.0,1834.0,2934.0,7201.0,2026,3,5,19
3,10,288894,137,288895,128,3.0,4,0,2988.0,266.0,...,16113.0,1401.0,169.0,1834.0,2934.0,7201.0,2026,3,5,19
4,10,288894,137,288895,128,3.0,2,0,2988.0,266.0,...,16113.0,1401.0,169.0,1834.0,2934.0,7201.0,2026,3,5,19
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
5564,110,233631,117,233632,215,3.0,1,1,3202.0,1420.0,...,3032.0,3836.0,1886.0,2588.0,1778.0,4567.0,2023,10,4,14
5565,110,233631,117,233632,215,3.0,6,1,3202.0,1420.0,...,3032.0,3836.0,1886.0,2588.0,1778.0,4567.0,2023,10,4,14
5566,110,233621,42,233622,68,3.0,6,0,553.0,39.0,...,457.0,739.0,1443.0,65.0,2049.0,209.0,2023,10,4,11
5567,110,233621,42,233622,68,3.0,3,1,553.0,39.0,...,457.0,739.0,1443.0,65.0,2049.0,209.0,2023,10,4,11


# Baseline accuracy rate

In [ ]:
mode = df.team1_win.mode()[0]
err_rate = len(df[df.team1_win == mode])/len(df)

0.5469563655952595

So the mode is right 55% of the time.